_studio/shared/umc/codec/h264_dec/src/umc_h264_dec_slice_decoder_decode_pic.cpp

```diff
@@ -74,11 +74,20 @@ Status H264Slice::UpdateReferenceList(ViewList &views,
     H264DecoderFrame *pLastInList[2] = {NULL, NULL};
 
     VM_ASSERT(m_pCurrentFrame);
-
-    pRefPicList0 = m_pCurrentFrame->GetRefPicList(m_iNumber, 0)->m_RefPicList;
-    pRefPicList1 = m_pCurrentFrame->GetRefPicList(m_iNumber, 1)->m_RefPicList;
-    pFields0 = m_pCurrentFrame->GetRefPicList(m_iNumber, 0)->m_Flags;
-    pFields1 = m_pCurrentFrame->GetRefPicList(m_iNumber, 1)->m_Flags;
+    if (m_pCurrentFrame == nullptr)
+        return UMC_ERR_NULL_PTR;
+
+    const H264DecoderRefPicList* pH264DecRefPicList0 = m_pCurrentFrame->GetRefPicList(m_iNumber, 0);
+    const H264DecoderRefPicList* pH264DecRefPicList1 = m_pCurrentFrame->GetRefPicList(m_iNumber, 1);
+    VM_ASSERT(pH264DecRefPicList0);
+    VM_ASSERT(pH264DecRefPicList1);
+    if (pH264DecRefPicList0 == nullptr || pH264DecRefPicList1 == nullptr)
```

Comment:
And if we have error being returned here, why do we need asserts:)? In such case constructions like:
`if (BUG_ON(condition)) return ...;`
are used. However, in this case I would probably just drop assert. We are not ready for the BUG_ON type of things I afraid.